### PR TITLE
Add ALZ Resource Types Script and Update Workflow

### PR DESCRIPTION
**Pull request by jtracey93**
# Overview/Summary

Add ALZ Resource Types Script and Update Workflow

## This PR fixes/adds/changes/removes

1. Add ALZ Resource Types Script and Update Workflow

### Breaking Changes

None

## Testing Evidence

Linting checks and workflow run itself will be sufficent
https://github.com/Azure/ALZ-Bicep/runs/7097377848?check_suite_focus=true#step:4:54

## As part of this Pull Request I have

- [x] Read the [Contribution Guide](https://github.com/Azure/ALZ-Bicep/wiki/Contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/ALZ-Bicep/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/ALZ-Bicep/issues)
- [ ] *(ALZ Bicep Core Team Only)* Associated it with relevant [ADO Items](https://aka.ms/alz/bicep/backlog)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/ALZ-Bicep/tree/main)
- [x] Performed testing and provided evidence.
- [ ] Updated one or more of the following tests *(if required)*
  - [Unit](https://github.com/Azure/ALZ-Bicep/blob/main/.github/workflows/bicep-build-to-validate.yml)
  - [Linting](https://github.com/Azure/ALZ-Bicep/tree/main/.github/workflows)
  - [E2E (End-To-End)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml)
  - [ValidateAzCloud (Base validation in Azure Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/base-unit-validate.yml)
  - [ValidateMcCloud (Base validation in Azure China Cloud)](https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/mc-base-unit-validate.yml)
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Module READMEs, Wiki Docs etc.)
- [ ] If relevant, created or updated Code Tours [here](https://github.com/Azure/ALZ-Bicep/blob/main/.vscode/tours)

### CODE REVIEW

File: .github/scripts/Get-AlzBicepResourceTypes.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Get-ChildItem -Path '.\infra-as-code\bicep\modules' -Recurse -Filter '*.bicep' -Exclude 'callModuleFromACR.example.bicep', 'orchHubSpoke.bicep' | ForEach-Object {
```
Comment (jfaurskov) [inline .github/scripts/Get-AlzBicepResourceTypes.ps1:1]: Not sure what this script is for if you are doing it as inline script in the workflow?

Comment (jtracey93) [reply .github/scripts/Get-AlzBicepResourceTypes.ps1:1]: Good spot. I left it here so can be used locally and we could convert the actions at some point in the future. 

Just making it modular and saving future work.

**Review by jfaurskov — CHANGES_REQUESTED** [review]
Wondering about what the script is for, as it doesn't look to be used anywhere?

**Comment by jtracey93** [comment]
> Wondering about what the script is for, as it doesn't look to be used anywhere?

Just replied to the comment 👍

Not used but there for the future if we decide we want that capability in the future. Also handy to run locally instead of having to take it from the workflow etc.

**Review by jfaurskov — APPROVED** [review]

**Pull request merged**